src/components/victory-chart.jsx

```diff
@@ -329,167 +94,241 @@ export default class VictoryChart extends React.Component {
+          } else {
+            counts[type] = counts[type] += 1;
+          }
+
+        },
+        limitReached: (child) => {
+          const type = child.type.displayName;
+          const axis = this.getAxisType(child);
+          if (!counts[type]) {
+            return false;
+          } else if (axis) {
+            return counts[type][axis] > 1;
+          } else if (_.includes(this.groupedDataTypes, type)) {
+            // TODO: should we remove the limit on grouped data types?
+            return counts[type] > 1;
+          }
+          return false;
+        },
+        total: (type, axis) => {
+          const totalCount = (axis && counts[type]) ?
+            counts[type][axis] : counts[type];
+          return totalCount || 0;
+        }
+      };
+    }();
+
+    const childComponents = [];
+    // loop through children, and add each child to the childComponents array
+    // unless the limit for that child type has already been reached.
+    React.Children.forEach(props.children, (child) => {
+      const type = child.type.displayName;
+      if (count.limitReached(child)) {
+        const msg = type === "VictoryAxis" ?
+          "Only one VictoryAxis component of each axis type is allowed when using the " +
+          "VictoryChart wrapper. Only the first axis will be used. Please compose " +
+          "multi-axis charts manually" :
+          "Only one " + type + "component is allowed per chart. If you are trying " +
+          "to plot several datasets, please pass an array of data arrays directly " +
+          "into " + type + ".";
+        log.warn(msg);
+      }
+      childComponents.push(child);
+      count.add(child);
+    });
+
+    // Add default axis components if necessary
+    // TODO: should we add both axes by default?
+    if (count.total("VictoryAxis", "x") < 1) { childComponents.push(defaultAxes.x); }
+    if (count.total("VictoryAxis", "y") < 1) { childComponents.push(defaultAxes.y); }
+
+    // Add defaut data if no data is provided
+    const dataComponents = _.filter(childComponents, (child) => {
+      return child.type.displayName !== "VictoryAxis";
+    });
+
+    if (dataComponents.length === 0) { childComponents.push(defaultData); }
+    return childComponents;
+  }
+
+  getDataComponents() {
+    return _.filter(this.childComponents, (child) => {
+      const type = child.type.displayName;
+      return !_.includes(this.groupedDataTypes, type) && type !== "VictoryAxis";
+    });
+  }
+
+  getGroupedDataComponents() {
+    return _.filter(this.childComponents, (child) => {
+      const type = child.type.displayName;
+      return _.includes(this.groupedDataTypes, type);
+    });
+  }
+
+  getAxisComponents(axis) {
+    return _.filter(this.childComponents, (child) => {
+      return axis === "x" ?
+      child.props.independentAxis && child.type.displayName === "VictoryAxis" :
+      child.type.displayName === "VictoryAxis" && !child.props.independentAxis;
+    });
+  }
```

Comment:
Hm, I don't like how much this whole set of methods depends on `displayName` being something specific. What if someone subclasses `VictoryAxis` and wants to use that? It just seems too fragile and specific to me.

A few options that come to mind:

```javascript
child instanceof VictoryAxis
```

or:

```javascript
class VictoryAxis extends React.Component {
  isAxis: true
  ...
}
```

or

```javascript
class VictoryAxis extends React.Component {
  role: "axis"
  ...
}
```

The latter is probably my favorite. Then every other Victory Component can get a `role` too and we'll be able to determine the "structure" of the chart (like we're doing in these methods) no matter which components are used (subclassed/customized/or not).